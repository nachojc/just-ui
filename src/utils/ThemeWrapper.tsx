import { ReactNode } from "react";
import { ThemeProvider } from "styled-components";
import theme from "../styles/themes";

type ThemeName = keyof typeof theme;

type options = {
  themeName: ThemeName;
};
export const wrapWithTheme = (children: ReactNode, options?: options) => {
  const opt: options = { ...{ themeName: "defaultTheme" }, ...options };
  return <ThemeProvider theme={theme[opt.themeName]}>{children}</ThemeProvider>;
};
