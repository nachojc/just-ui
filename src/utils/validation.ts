export const hasText = (value?: string): boolean => {
  return typeof value === "string" && value?.trim().length > 0;
};
