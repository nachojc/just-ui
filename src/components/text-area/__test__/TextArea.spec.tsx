import { fireEvent, render } from "@testing-library/react";
import { TextArea } from "..";
import React from "react";
import { wrapWithTheme } from "../../../utils/ThemeWrapper";

describe("TextArea", () => {
  it("default render", async () => {
    const { asFragment } = render(wrapWithTheme(<TextArea />));

    const btn = asFragment();
    expect(btn).toMatchSnapshot();
  });
  it("with Label", async () => {
    const { asFragment } = render(wrapWithTheme(<TextArea label="TextArea" />));
    const btn = asFragment();
    expect(btn).toMatchSnapshot();
  });
  it("with error", async () => {
    const { asFragment } = render(wrapWithTheme(<TextArea label="TextArea" error="no text" />));
    const btn = asFragment();
    expect(btn).toMatchSnapshot();
  });
  it("with display asBlock", async () => {
    const { asFragment } = render(wrapWithTheme(<TextArea asBlock={true} />));
    const btn = asFragment();
    expect(btn).toMatchSnapshot();
  });
  it("with disabled", async () => {
    const { asFragment } = render(wrapWithTheme(<TextArea disabled={true} />));
    const btn = asFragment();
    expect(btn).toMatchSnapshot();
  });
  it("small", async () => {
    const { asFragment } = render(wrapWithTheme(<TextArea size="small" />));
    const btn = asFragment();
    expect(btn).toMatchSnapshot();
  });
  it("focus", async () => {
    const handleOnFocus = jest.fn();
    const { getByTestId } = render(wrapWithTheme(<TextArea asBlock={true} onFocus={handleOnFocus} />));

    const label = getByTestId("TextArea-label");
    fireEvent.click(label);
    expect(handleOnFocus).toHaveBeenCalled();
  });
});
