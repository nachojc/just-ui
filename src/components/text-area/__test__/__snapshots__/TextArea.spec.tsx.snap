// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea default render 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb hBBJiZ"
  >
    <div
      class="sc-egkSDF leasZd"
    >
      <textarea
        class="sc-gtLWhw jpkqNe"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextArea small 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb hBBJiZ"
  >
    <div
      class="sc-egkSDF leasZd"
    >
      <textarea
        class="sc-gtLWhw hLxVsq"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextArea with Label 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb hBBJiZ"
  >
    <div
      class="sc-dntaoT hYNaTL"
      data-testid="TextArea-label"
    >
      TextArea
    </div>
    <div
      class="sc-egkSDF leasZd"
    >
      <textarea
        class="sc-gtLWhw jpkqNe"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextArea with disabled 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb hBBJiZ"
  >
    <div
      class="sc-egkSDF bUWDbo"
      disabled=""
    >
      <textarea
        class="sc-gtLWhw hHAYCT"
        disabled=""
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextArea with display asBlock 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb hBBJiZ"
  >
    <div
      class="sc-dntaoT hYNaTL"
      data-testid="TextArea-label"
    />
    <div
      class="sc-egkSDF leasZd"
    >
      <textarea
        class="sc-gtLWhw jpkqNe"
      />
    </div>
    <div
      class="sc-fAUdSK CeDQe"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea with error 1`] = `
<DocumentFragment>
  <div
    class="sc-blHHSb hBBJiZ"
  >
    <div
      class="sc-dntaoT hYNaTL"
      data-testid="TextArea-label"
      error="no text"
    >
      TextArea
    </div>
    <div
      class="sc-egkSDF bklFls"
      error="no text"
    >
      <textarea
        class="sc-gtLWhw jpkqNe"
        error="no text"
      />
    </div>
    <div
      class="sc-fAUdSK iJytMv"
      error="no text"
    >
      no text
    </div>
  </div>
</DocumentFragment>
`;
